bench/macro/lsm-tree-bench-unions.hs

```diff
@@ -0,0 +1,629 @@
+import           System.Mem (performMajorGC)
+import qualified System.Random as Random
+import           Text.Printf (printf)
+import qualified Text.Read as Read
+
+import           Database.LSMTree.Extras (groupsOfN)
+import qualified Database.LSMTree.Extras.Random as Random
+import           Database.LSMTree.Internal.ByteString (byteArrayToSBS)
+
+-- We should be able to write this benchmark
+-- using only use public lsm-tree interface
+import qualified Database.LSMTree.Simple as LSM
+
+-------------------------------------------------------------------------------
+-- Constant Values
+-------------------------------------------------------------------------------
+
+baselineTableID :: Int
+baselineTableID = 0
+
+baselineTableName :: LSM.SnapshotName
+baselineTableName = makeTableName baselineTableID
+
+-------------------------------------------------------------------------------
+-- Keys and values
+-------------------------------------------------------------------------------
+
+type K = BS.ShortByteString
+type V = BS.ShortByteString
+
+label :: LSM.SnapshotLabel
+label = LSM.SnapshotLabel "K V B"
+
+-- | We generate 34 byte keys by using a PRNG to extend a word64 to 32 bytes
+-- and then appending two constant bytes. This corresponds relatively closely
+-- to UTxO keys, which are 32 byte cryptographic hashes, followed by two bytes
+-- which are typically the 16bit value 0 or 1 (a transaction output index).
+--
+makeKey :: Word64 -> K
+makeKey seed =
+    case P.runPrimArray $ do
+           v <- P.newPrimArray 5
+           let g0 = Random.mkStdGen (fromIntegral seed)
+           let (!w0, !g1) = Random.uniform g0
+           P.writePrimArray v 0 w0
+           let (!w1, !g2) = Random.uniform g1
+           P.writePrimArray v 1 w1
+           let (!w2, !g3) = Random.uniform g2
+           P.writePrimArray v 2 w2
+           let (!w3, _g4) = Random.uniform g3
+           P.writePrimArray v 3 w3
+           P.writePrimArray v 4 0x3d3d3d3d3d3d3d3d -- ========
+           case v of
+             P.MutablePrimArray mba -> do
+               _ <- P.resizeMutableByteArray (P.MutableByteArray mba) 34
+               return v
+
+      of (P.PrimArray ba :: P.PrimArray Word64) ->
+           byteArrayToSBS (P.ByteArray ba)
+
+-- We use constant value. This shouldn't affect anything.
+theValue :: V
+theValue = BS.replicate 60 120 -- 'x'
+{-# NOINLINE theValue #-}
+
+-------------------------------------------------------------------------------
+-- Options and commands
+-------------------------------------------------------------------------------
+
+data GlobalOpts = GlobalOpts
+    { rootDir     :: !FilePath -- ^ session directory.
+    , tableCount  :: !Int   -- ^ Number of  tables in the benchmark
+    , initialSize :: !Int  -- ^ Initial size of each table in the benchmark
+    , seed        :: !(Maybe Int) -- ^ The seed for the RNG for deterministic behavior,
+                           --   use system entropy when not explicitly provided.
```

Comment:
I'd use a fixed seed as the default value, just some arbitrary integer that you pick. If we use system entropy, then the benchmark will be harder to reproduce

Reply:
Reverted back to using a default seed value.